src/core/cluster/quorum.rs

```diff
@@ -14,99 +16,187 @@ impl Quorum {
             quorum,
         }
     }
-}
-
-#[async_trait]
-impl Cluster for Quorum {
-    async fn put(&self, key: BobKey, data: BobData) -> Result<(), Error> {
-        debug!("get nodes of the target vdisk");
-        let target_nodes = get_target_nodes(&self.mapper, key);
-        debug!("PUT[{}]: Nodes for fan out: {:?}", key, &target_nodes);
-        debug!("call put on target nodes (on target vdisk)");
-        let results = LinkManager::call_nodes(target_nodes.iter(), |conn| {
-            Box::pin(conn.put(key, data.clone(), PutOptions::new_local()))
-        })
-        .await;
-        debug!("PUT[{}] rcv {} cluster answers", key, results.len());
 
-        let total_ops = results.len();
-        debug!("filter out ok results");
-        let errors = results
-            .into_iter()
-            .filter_map(Result::err)
+    async fn put_at_least(&self, key: BobKey, data: BobData) -> Result<(), Error> {
+        debug!("PUT[{}] ~~~PUT LOCAL NODE FIRST~~~", key);
+        let mut local_put_ok = 0_usize;
+        let mut remote_ok_count = 0_usize;
+        let mut at_least = self.quorum;
+        let (vdisk_id, disk_path) = self.mapper.get_operation(key);
+        if let Some(path) = disk_path {
```

Comment:
may be put local put logic to function?

Reply:
It won't be nicer) Logic was already put to `put_local_all`